### PR TITLE
Content update - Dependency properties overview (user story 1878471)

### PR DESCRIPTION
**Pull request by v-trisshores**
Hi @adegeo, this PR is ready for your review (article and code projects).

Notes:

Some of the links point to .NET Framework 4.8 and others to .NET 5.0 - all are hardcoded per your [previous instructions](https://github.com/dotnet/docs-desktop/pull/1141#discussion_r697791616).

However, if we don't hardcode the .NET version when linking to framework articles, the server redirects to the latest version. For example:

/dotnet/desktop/wpf/advanced/xaml-in-wpf?view=netframeworkdesktop-4.8&preserve-view=true  (will only point to the .NET 4.8 page)

/dotnet/desktop/wpf/advanced/xaml-in-wpf  (will auto-redirect to the .NET 5.0 page)

[Internal preview link](https://review.docs.microsoft.com/en-us/dotnet/desktop/wpf/properties/dependency-properties-overview?view=netdesktop-5.0&branch=pr-en-us-1160)

Contributes to #1155

### CODE REVIEW

**Comment by adegeo** [comment]
If the page is available in .NET 5, you should link to the .NET 5 version.

**Comment by v-trisshores** [comment]
I'll check all links in the 3 print articles and submit a separate PR for any issues found. I'm glad we're hashing this out early!

Conditional link update should be automatable, like image/snippet syntax update in docs-toolset.

**Review by adegeo — APPROVED** [review]
LGTM, just some fixes for the links. I'll put this in the TOC and create the redirect.

**Comment by adegeo** [comment]
Another thing I was thinking of, if you run into XAML that happens to be different between VB and C# (generally just the root nodes are different, such as Window or Class because VB does it differently) I call out that the XAML may look different in Visual Basic and all XAML is from C#. There isn't any of that here, but I just want you to be aware of that 😄

File: dotnet-desktop-guide/net/wpf/documents/dependency-properties-overview.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+Windows Presentation Foundation (WPF) provides a set of services that can be used to extend the functionality of a type's [property](/dotnet/standard/base-types/common-type-system#properties). Collectively, these services are referred to as the WPF property system. A property that's backed by the WPF property system is known as a dependency property. This overview describes the WPF property system and the capabilities of a dependency property, including how to use existing dependency properties in XAML and in code. This overview also introduces specialized aspects of dependency properties, such as dependency property metadata, and how to create your own dependency property in a custom class.
+
+[!INCLUDE [desktop guide under construction](../../includes/desktop-guide-preview-note.md)]
+
+## Prerequisites
+
+This article assumes basic knowledge of the .NET type system and object-oriented programming. To follow the examples in this article, it helps to understand XAML and know how to write WPF applications. For more information, see [Walkthrough: My first WPF desktop application](/dotnet/desktop/wpf/get-started/create-app-visual-studio?view=netdesktop-5.0&preserve-view=true).
```
[the remark this answers is not in the document]
Comment (v-trisshores) [reply dotnet-desktop-guide/net/wpf/documents/dependency-properties-overview.md]: Done.

**Comment by v-trisshores** [comment]
@adegeo thanks for the clarification on links, which are now fixed in this article.

I'll keep an eye out for any XAML changes between C# & VB for future articles.

**Review by adegeo — CHANGES_REQUESTED** [review]
Actually, I just realized this is in the wrong folder. This should be in a `wpf/properties` folder, not the `wpf/documents` folder.

**Comment by v-trisshores** [comment]
@adegeo, I've relocated the article & snippets to wpf/properties folder. 

Should we rename wpf/documents to wpf/printing?

**Comment by v-trisshores** [comment]
One of the other projects I worked on, had articles marked with this kind of comment below the metadata section:


Would you be interested in that score getting added for future reference?

**Comment by adegeo** [comment]
> @adegeo, I've relocated the article & snippets to wpf/properties folder.
> 
> Should we rename wpf/documents to wpf/printing?

Nah, the printing stuff is highly related to the overall documents story which will contain other things. Originally I was thinking this article would go in the `systems` folder since it's related to the WPF systems (properties, events, resources) but I think there will end up being a lot in that folder and I want to avoid the situation where 100 articles live in the same generic folder like the `advanced` folder in the .NET Framework content.

> @adegeo, a different project I worked on had articles marked with this kind of comment below the metadata section:
> 
> 
> Would you be interested in that score getting added for future reference?

Sure that sounds good!

**Comment by adegeo** [comment]
@v-trisshores Here is the PR for you to accept to get the TOC/Redirects into your branch for this PR... woah that is a lot of PRs... https://github.com/v-trisshores/docs-desktop/pull/1

**Review by adegeo — APPROVED** [review]

**Comment by adegeo** [comment]
Awesome thanks again!

**Comment by v-trisshores** [comment]
Glad to help!

**Pull request merged**